Doc/c-api/long.rst

```diff
@@ -368,36 +368,77 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
           // Success!
       }
       else {
-          // Overflow occurred, but 'bignum' contains a truncated value.
+          // Overflow occurred, but 'value' contains the truncated
+          // lowest bits of pylong.
       }
 
+   The above example may look *similar* to
+   :c:func:`PyLong_As* <PyLong_AsSize_t>`
+   but instead fills in a specific caller defined type and never raises an
+   error about of the :class:`int` *pylong*'s value regardless of *n_bytes*
+   or the returned byte count.
+
+   To get at the entire potentially big Python value, this can be used to
+   reserve enough space and copy it::
+
+      // Ask how much space we need.
+      Py_ssize_t expected = PyLong_AsNativeBits(pylong, NULL, 0, -1);
+      if (expected < 0) {
+          // A Python exception was set with the reason.
+          return NULL;
+      }
+      assert(expected != 0);  // Impossible per the API definition.
+      uint8_t *bignum = malloc(expected);
+      if (!bignum) {
+          PyErr_SetString(PyExc_MemoryError, "bignum malloc failed.");
+          return NULL;
+      }
+      // Safely get the entire value.
+      Py_ssize_t bytes = PyLong_AsNativeBits(pylong, bignum, expected, -1);
+      if (bytes < 0 || bytes > expected) {  // Be safe, should not be possible.
+          if (!PyErr_Occurred()) {
+              PyErr_SetString(PyExc_RuntimeError,
+                  "Unexpected bignum truncation after a size check.");
+          }
+          free(bignum);
+          return NULL;
+      }
+      // The expected success given the above pre-check.
+      // ... use bignum ...
+      free(bignum);
+
    *endianness* may be passed ``-1`` for the native endian that CPython was
    compiled with, or ``0`` for big endian and ``1`` for little.
 
-   Return ``-1`` with an exception raised if *pylong* cannot be interpreted as
+   Returns ``-1`` with an exception raised if *pylong* cannot be interpreted as
```

Comment:
Docs team has apparently decided to use `Return` in this context. I much prefer `Returns`, but I'm not part of that group so I don't know the rationale.

Reply:
Noted. Returns is correct English tense when describing what something does rather than writing code telling a system what to do.  Our docs are not a spec or code, and even a spec would preceed the word return with a qualifier like should or must. :)  I'm going to leave it and let them come through and reword things later for global consistence if they feel otherwise.